### PR TITLE
[A2-865] Add db_id to members table

### PR DESCRIPTION
**Pull request by blakestier**
### :athletic_shoe: Demo Script / Repro Steps

bring up dev env/`start_all_services`
Add, delete, update members in UI.

### :white_check_mark: Checklist

- [ ] Necessary tests added/updated?
- [ ] Necessary docs added/updated?
- [ ] Code actually executed?
- [ ] Vetting performed (unit tests, lint, etc.)?

### CODE REVIEW

File: components/authz-service/storage/postgres/migration/sql/52_add_db_id_members.up.sql [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    RETURN (
+        -- no projects filter requested (length 0) will be the case for v2.0 or v2.1 ["*"]
+        array_length(_projects_filter, 1) IS NULL
+        -- projects filter intersects with projects for row
+        OR _projects && _projects_filter
+        -- projects for row is an empty array, check if (unassigned) in project filter
+        OR (array_length(_projects, 1) IS NULL
+            AND '{(unassigned)}' && _projects_filter));
+END
+$$
```
Comment (msorens) [inline components/authz-service/storage/postgres/migration/sql/52_add_db_id_members.up.sql]: I don't see any code changed in this function, just formatting...?

File: components/authz-service/storage/postgres/migration/sql/52_add_db_id_members.up.sql [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+                        stmt.resources,
+                        stmt.role,
+                        -- get each statement's projects by cross-referencing iam_policy_statements and iam_statement_projects
+                        (
+                            SELECT
+                                COALESCE(Json_agg(proj.id) FILTER (WHERE proj.id IS NOT NULL),
+                                    '[]')
+                                FROM iam_statement_projects AS stmt_projs
+                            LEFT OUTER JOIN iam_projects AS proj ON stmt_projs.statement_id = stmt.id WHERE stmt_projs.project_id = proj.id) AS projects FROM iam_policy_statements AS pol_stmts
+                    LEFT OUTER JOIN iam_statements AS stmt ON pol_stmts.statement_id = stmt.id WHERE pol_stmts.policy_id = pol.id
```
Comment (msorens) [inline components/authz-service/storage/postgres/migration/sql/52_add_db_id_members.up.sql]: The formatting changes particularly on these couple lines above make it very difficult and error prone to read.

File: components/authz-service/storage/postgres/migration/sql/52_add_db_id_members.up.sql [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                    stmt.id)
+            SELECT
+                array_agg(statement_rows) FILTER (WHERE statement_rows.id IS NOT NULL)
+            FROM statement_rows) AS statements,
+            -- get policy members
+            (
+            SELECT
+                array_agg(mem) FILTER (WHERE mem.id IS NOT NULL)
+            FROM iam_policy_members AS pol_mems
+        LEFT OUTER JOIN iam_members AS mem ON pol_mems.member_id = mem.db_id WHERE pol_mems.policy_id = pol.id) AS members,
```
Comment (msorens) [inline components/authz-service/storage/postgres/migration/sql/52_add_db_id_members.up.sql]: More formatting problems: having two "AS" on one line is a major warning sign to me.

File: components/authz-service/storage/postgres/migration/sql/52_add_db_id_members.up.sql [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+            (
+            SELECT
+                array_agg(proj.id) FILTER (WHERE proj.id IS NOT NULL)
+            FROM iam_policy_projects AS pol_projs
+            LEFT OUTER JOIN iam_projects AS proj ON pol_projs.project_id = proj.id WHERE pol_projs.policy_id = pol.id) AS projects FROM iam_policies AS pol WHERE pol.id = _policy_id
+        GROUP BY
+            pol.id
+)
+SELECT
+    json_build_object('id', temp.id, 'name', temp.name, 'type', temp.type, 'statements', COALESCE(temp.statements, '{}'), 'members', COALESCE(temp.members, '{}'), 'projects', COALESCE(temp.projects, '{}')) AS POLICY
```
Comment (msorens) [inline components/authz-service/storage/postgres/migration/sql/52_add_db_id_members.up.sql]: Should be one item per line; line is too long here.

**Comment by blakestier** [comment]
closing in favor of https://github.com/chef/automate/pull/748

**Pull request closed without merging**